src/common/spawn.c

```diff
@@ -63,3 +79,87 @@ spawn_async_no_shell(char const *command)
 	g_strfreev(argv);
 }
 
+pid_t
+spawn_piped(const char *command, int *pipe_fd)
+{
+	assert(command);
+
+	int pipe_rw[2];
+	if (pipe(pipe_rw) != 0) {
+		wlr_log(WLR_ERROR, "unable to pipe()");
+		return -1;
+	}
+
+	pid_t pid = fork();
+	if (pid < 0) {
+		close(pipe_rw[0]);
+		close(pipe_rw[1]);
+		wlr_log(WLR_ERROR, "unable to fork()");
+		return pid;
+	}
+
+	if (pid == 0) {
+		/* child */
+		reset_signals_and_limits();
+
+		/*
+		 * replace stdin and stderr with /dev/null
+		 * and stdout with the write end of the pipe
+		 */
+		dup2(pipe_rw[1], STDOUT_FILENO);
+
+		int dev_null = open("/dev/null", O_RDWR);
+		if (dev_null < 1) {
+			perror("opening /dev/null failed");
+			/*
+			 * Just close stdin and stderr and
+			 * hope $command can deal with that.
+			 */
+			close(STDIN_FILENO);
+			close(STDERR_FILENO);
+		} else {
+			dup2(dev_null, STDIN_FILENO);
+			dup2(dev_null, STDERR_FILENO);
+			close(dev_null);
+		}
+
+		/* close all remaining fds */
+		close(pipe_rw[0]);
+		close(pipe_rw[1]);
+
+		execl("/bin/sh", "sh", "-c", command, NULL);
+		perror("execl");
+		_exit(1);
+	}
+
+	/* labwc */
+
+	/*
+	 * Prevent leaking the read end of the pipe to further
+	 * children forked during the lifetime of the descriptor.
+	 */
+	set_cloexec(pipe_rw[0]);
+
+	close(pipe_rw[1]);
+	*pipe_fd = pipe_rw[0];
+	return pid;
+}
+
+void
+spawn_piped_close(pid_t pid, int pipe_fd)
+{
+	close(pipe_fd);
+
+	int details = 0;
+	int options = 0;
+	if (waitpid(pid, &details, options) == -1) {
+		perror("waitpid failed");
+		return;
+	}
```

Comment:
evaluate using non-blocking waitpid and a fallback labwc wide SIGCHLD handler within the wl event loop

Reply:
Uses non-blocking now but labwc wide SIGCHLD handler still missing.

Reply:
added the labwc wide SIGCHLD handler.